src/jdk.jfr/share/classes/jdk/jfr/internal/instrument/FileChannelImplInstrumentor.java

```diff
@@ -66,22 +68,30 @@ public void force(boolean metaData) throws IOException {
     @JIInstrumentationMethod
     public int read(ByteBuffer dst) throws IOException {
         FileReadEvent event = FileReadEvent.EVENT.get();
-        if (!event.isEnabled()) {
-            return read(dst);
-        }
+        FileReadIOStatisticsEvent readPeriodicEvent = FileReadIOStatisticsEvent.EVENT.get();
         int bytesRead = 0;
-        try {
-            event.begin();
+
+        long startTime = System.nanoTime();
+        if (!event.isEnabled()) {
             bytesRead = read(dst);
-        } finally {
-            if (bytesRead < 0) {
-                event.endOfFile = true;
-            } else {
-                event.bytesRead = bytesRead;
+        } else if (event.isEnabled()) {
```

Comment:
this else condition is redundant, the compiler should get rid of it, but we don't need to call isEnabled() twice [same in the other 3 files]

Reply:
I agree. I've removed the unnecessary 'else if' condition.